torch/distributed/_tensor/sharding_prop.py

```diff
@@ -17,19 +18,19 @@ def unwrap_schema(e: object) -> object:
 
 class ShardingPropagator(object):
     def __init__(self) -> None:
-        self.op_to_rules: Dict[str, Callable[[OpSchema], OutputSharding]] = {}
+        self.op_to_rules: Dict[OpOverload, Callable[[OpSchema], OutputSharding]] = {}
 
     def register_sharding_prop_rule(
-        self, op_key: str, rule_func: Callable[[OpSchema], OutputSharding]
+        self, op_overload: OpOverload, rule_func: Callable[[OpSchema], OutputSharding]
```

Comment:
Does MyPy do the type check down the road? What if user pass in a valid op?

Reply:
you mean an invalid op? it would trigger an error i believe, mypy would probably also complain if user set it up correctly

Reply:
Got it, thanks!